vtex-apps/b2b-organizations-graphql
# fix: add analytics client to start sending metrics again

#### What problem is this solving?
Metrics were not being emitted in prod (only in local workspaces) due to the following error:
```
Error to send metrics from auth metric
...
{"__VTEX_IO_LOG":true,"level":"error","app":"vtex.b2b-organizations-graphql@0.48.1","account":"rahrqa","workspace":"master","production":true,"data":{"error":{"name":"Error","message":"Client network socket disconnected before secure TLS connection was established - after 2 retries.","stack":"Error: Client network socket disconnected before secure TLS connection was established - after 2 retries.\n    at Object.exports.sendMetric (/usr/src/node/utils/metrics/metrics.ts:24:13)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)\n    at Object.exports.sendMetric (/usr/src/node/utils/metrics/metrics.ts:22:7)\n    at Object.exports.sendMetric (/usr/src/node/utils/metrics/metrics.ts:22:7)\n    at Object.sendAuthMetric [as default] (/usr/src/node/utils/metrics/auth.ts:34:5)\n    at AuditAccess.sendAuthMetric (/usr/src/node/resolvers/directives/auditAccess.ts:53:5)"},"message":"Error to send metrics from auth metric"},"operationId":"ad1f1888-0b61-47ff-afd3-2293c3f1d1c0","requestId":"6e56b7c241384a0b82edb9f14a75f98f"}
```

After discussing with #analytics-support they suggested implementing a client using the IOClient that already exists.

Locally the solution worked. But we need to deploy to production to make sure it also works there:
![Captura de Tela 2024-02-20 às 13 21 55](https://github.com/vtex-apps/b2b-organizations-graphql/assets/131273915/9c15a1c9-8b1a-4850-941c-0bf16418318c)

#### How to test it?

Go to the Admin Organizations page, open an Organization, go to the users tab, add an user and remove it. You should have some related metrics on redshift after a couple of hours.

[Workspace](https://b2bteam1558--b2bstoreqa.myvtex.com/admin/b2b-organizations)
